packages/flutter/lib/src/material/dialog.dart

```diff
@@ -481,12 +486,15 @@ class _DialogRoute<T> extends PopupRoute<T> {
 Future<T> showDialog<T>({
   @required BuildContext context,
   bool barrierDismissible: true,
-  @required Widget child,
+  @deprecated Widget child,
+  @required WidgetBuilder builder,
```

Comment:
builder can't be required while child is still present, since it's valid to only provide child in that case.

Reply:
Done